carla-carma-integration/carla_carma_bridge/src/carla_carma_bridge/carma_carla_route

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+# Copyright (C) 2021 LEIDOS.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+"""
+Call Services from CARMA: 
+    Service: /guidance/get_available_routes
+             /guidance/set_active_route
+
+Publish to CARMA :cav_msgs::Route
+    
+"""
+
+import rospy
+from geometry_msgs.msg import PoseStamped, Pose
+from cav_msgs.msg import RouteState
+from cav_msgs.msg import Route
+from cav_msgs.msg import RouteEvent
+from cav_srvs.srv import GetAvailableRoutes
+from cav_srvs.srv import SetActiveRoute
+
+selected_route = rospy.get_param('selected_route')
+received_pose = False
+
+class ErrorStatus:
+    def __init__(self, value, text):
+        self.value = value
+        self.text = text
+        
+NO_ERROR = ErrorStatus(0, "NO_ERROR")
+NO_ROUTE = ErrorStatus(1, "NO_ROUTE")
+ALREADY_FOLLOWING_ROUTE = ErrorStatus(2, "ALREADY_FOLLOWING_ROUTE")
+ROUTE_FILE_ERROR = ErrorStatus(3, "ROUTE_FILE_ERROR")
+ROUTING_FAILURE = ErrorStatus(4, "ROUTING_FAILURE")
+TRANSFORM_ERROR = ErrorStatus(5, "TRANSFORM_ERROR")
+
+#Get the routes to be set after localization
+def get_available_routes():
+    service = rospy.ServiceProxy('/get_available_routes', GetAvailableRoutes)     
+    try:
+        resp = service(0)
+        available_routes = GetAvailableRoutes()
+        available_routes.availableRoutes = resp.availableRoutes
+
+    except:
+        print("Invalid: /guidance/get_available_routes failed.")
+    else:
+        print("Available Routes acquired")
+        return available_routes.availableRoutes
+
+def set_route(id):
+    service = rospy.ServiceProxy('/set_active_route', SetActiveRoute)
+
+    try:
+        #Service Call Request
+        resp = service(0, id)
+        if resp.errorStatus != NO_ERROR.value:
+            if resp.errorStatus == NO_ROUTE.value:
```

Comment:
More what I was referring to was I don't understand the purpose of these internal if-statements and checks. You've removed the `errorDescription` variable (though it's referenced below in a few places so it still exists), but  as far as I can tell these checks on the error code don't have any function. I think it should be fine to just check `if there was an error: log/warn/etc. else: proceed with execution`